model/commitq/githubpr/github_pr_sender.go

```diff
@@ -0,0 +1,97 @@
+package githubpr
+
+import (
+	"context"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/evergreen-ci/evergreen/model/commitq"
+	"github.com/google/go-github/github"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+type githubPRLogger struct {
+	prService    *github.PullRequestsService
+	statusSender send.Sender
+	*send.Base
+}
+
+// NewGithubPRLogger creates a new Sender implementation that
+// merges a pull request
+// Specify an OAuth token for GitHub authentication
+func NewGithubPRLogger(name string, token string, statusSender send.Sender) (send.Sender, error) {
+	ctx := context.TODO()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	githubClient := github.NewClient(tc)
+	s := &githubPRLogger{
+		Base:         send.NewBase(name),
+		prService:    githubClient.GetPullRequests(),
+		statusSender: statusSender,
+	}
+
+	fallback := log.New(os.Stdout, "", log.LstdFlags)
+	if err := s.SetErrorHandler(send.ErrorHandlerFromLogger(fallback)); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (s *githubPRLogger) Send(m message.Composer) {
+	if !s.Level().ShouldLog(m) {
+		return
+	}
+
+	msg := m.Raw().(*GithubMergePR)
```

Comment:
I would avoid doing unchecked casts like this.

Reply:
(The way to do this is `msg, ok := m.Raw().(*GithubMergePR)` and return an error if `ok != true`.)